docs/application/dotnet/guides/app-management/application-launcher.md

```diff
@@ -0,0 +1,302 @@
+    {
+        AppId = id;
+    }
+}
+```
+
+4. The `ApplicationIcon` class stores  the application id. The `Icon` component and `OriginSize` are used to resize `Icon` when it is in the pressed state. The `ApplicationIconClicked` is invoked when touch changes its state to finished:
+
+```csharp
+class ApplicationIcon : View
+{
+    private string AppId;
+
+    private ImageView Icon;
+    private Size2D OriginSize;
+
+    public event EventHandler<ApplicationIconClickedEventArgs> ApplicationIconClicked;
+
+    //...
+}
+```
+
+5. The `ApplicationIcon` constructor is responsible for:
+  - Set `AppId` and `OriginSize`
+  - Create view components: `Label` which is used to show an application name and `Icon` which load resource from `path` string and show loaded image.
+  - Create `Layout` of the `ApplicationIcon`. In this case, the vertical linear layout is used.
+  - Setup the `TouchEvent` handler:
+
+```csharp
+public ApplicationIcon(string name, string path, Size2D size, string id)
+{
+    this.AppId = id;
+    OriginSize = new Size2D(size.Width - 40, size.Height - 40);
+
+    this.Size2D = size;
+
+    TextLabel label = new TextLabel()
+    {
+        PointSize = 6,
+        Text = name,
+        MultiLine = true,
+        Size2D = new Size2D(size.Width, 40),
+        HorizontalAlignment = HorizontalAlignment.Center,
+        VerticalAlignment = VerticalAlignment.Center,
+    };
+
+    Icon = new ImageView()
+    {
+        Size2D = new Size2D(size.Height - 40, size.Height - 40),
+        ResourceUrl = path,
+    };
+
+    this.Layout = new LinearLayout()
+    {
+        LinearAlignment = LinearLayout.Alignment.CenterHorizontal,
+        LinearOrientation = LinearLayout.Orientation.Vertical
+    };
+
+    this.Add(Icon);
+    this.Add(label);
+
+    this.TouchEvent += OnTouchEvent;
+}
+```
+
+**Figure: Application Icons**
+
+![Application Icons](./media/application_launcher_icons.png)
+
+6. `OnTouchEvent` reads state from the `TouchEventArgs`. If touch is in `PointStateType.Down` state, the application icon is resized. Otherwise, it returns to normal size and predefined event is invoked with the proper `AppId`:
+
+```csharp
+public bool OnTouchEvent(object sender, TouchEventArgs args)
+{
+    var state = args.Touch.GetState(0);
+
+    switch (state)
+    {
+        case PointStateType.Down:
+            Icon.Size2D = new Size2D((int)(OriginSize.Width * 1.1f), (int)(OriginSize.Height * 1.1f));
+            break;
+        case PointStateType.Finished:
+            Icon.Size2D = OriginSize;
+            ApplicationIconClicked.Invoke(this, new ApplicationIconClickedEventArgs(this.AppId));
+            break;
+        default:
+            break;
+    }
+
+    return false;
+}
+```
+
+7. The `Program` class derived from NUIApplication handles all necessary system events ([Application Lifecycle](./01_application_lifecycle.md)). The `AppLauncher` is created in `Initialize()` method and used in an icon touch handler:
```

Comment:
Global. Would you please check the file names again? The links are broken. I also would  like to recommend you to fetch the latest version and rebase.